dask/array/backends.py

```diff
@@ -296,3 +300,154 @@ def _nannumel_sparse(x, **kwargs):
+
+
+class ArrayBackendEntrypoint(DaskBackendEntrypoint):
+    """Dask-Array version of ``DaskBackendEntrypoint``
+
+    See Also
+    --------
+    NumpyBackendEntrypoint
+    """
+
+    def __init__(self):
+        """Register data-directed dispatch functions"""
+        raise NotImplementedError
+
+    @property
+    def RandomState(self):
+        """Return the backend-specific RandomState class
+
+        For example, the 'numpy' backend simply returns
+        ``numpy.random.RandomState``.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def ones(shape, *, dtype=None, meta=None):
+        """Create an array of ones
+
+        Returns a new array having a specified shape and filled
+        with ones.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def zeros(shape, *, dtype=None, meta=None):
+        """Create an array of zeros
+
+        Returns a new array having a specified shape and filled
+        with zeros.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def empty(shape, *, dtype=None, meta=None):
+        """Create an empty array
+
+        Returns an uninitialized array having a specified shape.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def full(shape, fill_value, *, dtype=None, meta=None):
+        """Create a uniformly filled array
+
+        Returns a new array having a specified shape and filled
+        with fill_value.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def arange(start, /, stop=None, step=1, *, dtype=None, meta=None):
+        """Create an ascending or descending array
+
+        Returns evenly spaced values within the half-open interval
+        ``[start, stop)`` as a one-dimensional array.
+        """
+        raise NotImplementedError
+
+
+class NumpyBackendEntrypoint(ArrayBackendEntrypoint):
+    def __init__(self):
+        # Data dispatch functions are already registered
+        # in the current module
+        pass
+
+    @property
+    def RandomState(self):
+        return np.random.RandomState
+
+
+class ArrayCreationDispatch(CreationDispatch):
+    def register_backend(
+        self, name: str, backend: BackendEntrypointType
+    ) -> BackendEntrypointType:
+        """Register a target class for a specific array-backend label"""
+        if not isinstance(backend, ArrayBackendEntrypoint):
+            raise ValueError(
+                f"ArrayCreationDispatch only supports "
+                f"ArrayBackendEntrypoint registration. "
+                f"Got {type(backend)}"
+            )
+        self._lookup[name] = backend
+        return cast(BackendEntrypointType, backend)
+
+
+array_creation_dispatch = ArrayCreationDispatch(
+    module_name="array",
+    default="numpy",
+    name="array_creation_dispatch",
+)
+
+
+array_creation_dispatch.register_backend("numpy", NumpyBackendEntrypoint())
+
+
+try:
+    import cupy
```

Comment:
Is there a version of this where this registration is the responsibility of `cupy`, and this whole `try`/`except` block goes away? I suppose this is nice in that you get registration without the user needing to explicitly import `cupy`, but it otherwise seems like something that might be better in that package.

Given the entrypoints-based approach, could dask proceed with backend discovery there, and auto-register any backend entrypoint it finds? That might avoid any extra user imports.

Reply:
>Is there a version of this where this registration is the responsibility of cupy, and this whole try/except block goes away?

Yes, the cupy entrypoint could certainly be defined in cupy. However, that would require me and/or others to submit PRs to a seperate project. I think we definitely want the keep things close while we iron out the kinks.

>Given the entrypoints-based approach, could dask proceed with backend discovery there, and auto-register any backend entrypoint it finds? That might avoid any extra user imports.

I'll admit that I completely forgot that I was making the `import cupy` attempt unavoidable here. Would it be resonable if I defined `CupyBackendEntrypoint` in a distinct file and exposed it as a "proper" entrypoint in `setup.cfg`/`setup.py`?

Reply:
> I'll admit that I completely forgot that I was making the import cupy attempt unavoidable here.

How expensive of an import is `cupy`? In #9230 we found that accidentally doing an unconditional import of `IPython` was visible in startup times for the `distributed` test suite.

Reply:
The latest code removes this import and exposes the cupy entrypoint in a distinct file - Let me know what you think.

Reply:
I checked this out locally, and it seems to work well!

Reply:
And by work well, I mean that I got an import error when I tried to set the backend to `cupy`, and not earlier.